### ISSUE TITLE
Callbacks mid-epoch

### ISSUE THREAD

**Issue by tetmin** [fchollet/keras#3330]
It doesn't seem possible to run callbacks after a certain number of iterations or batches, only at the end of each epoch, is that correct?

I'd like to monitor validation loss during an epoch (since they are slow epochs). I'd also like to be able to store a moving average of model weights as is possible with Tensorflow, this usually produces significantly better results than using a single snapshot.

**Comment by ncullen93** [comment]
not correct. use the 'on_batch_end()' or 'on_batch_begin()' method and keep a counter of which batch it is.. run whatever you want to run when the counter reaches the correct number.
